src/ssl.c

```diff
@@ -52853,15 +52938,21 @@ void wolfSSL_X509_REQ_free(WOLFSSL_X509* req)
 int wolfSSL_X509_REQ_sign(WOLFSSL_X509 *req, WOLFSSL_EVP_PKEY *pkey,
                           const WOLFSSL_EVP_MD *md)
 {
+    int ret;
```

Comment:
This hard coded `der` size of 2048 is not great. I know this is existing code, but can we improve this? Looks like 10 other places in ssl.c and internal.c with this pattern. If you want to followup with a new PR that is fine too. Everything else in this PR is good.

Either use an enum/macro or allocate from heap based on what we know from `WOLFSSL_X509* req`.

Reply:
I'd like to do a follow up PR with this improvement. Just to get the wolfCLU API needed into master first.

Reply:
Just saw looks like there is a merge conflict. Can add the improvement when resolving the merge conflict since the tests will need to rerun anyway.